src/FileSystemUtilities.js

```diff
@@ -63,6 +64,12 @@ export default class FileSystemUtilities {
       }
       type = "file";
     }
+    if (pathIsAbsolute(src) && process.platform !== "win32") {
+      if (type === "junction") {
+        type = "dir";
+      }
```

Comment:
You're totally right, I must have misread that. For some reason I had it in my head that only `'junction'` was ignored on other platforms, but it says all of them are. Makes total sense. I'll push a fix.